sympy/functions/combinatorial/factorials.py

```diff
@@ -577,6 +577,10 @@ def eval(cls, x, k):
                             return 1/reduce(lambda r, i:
                                             r*(x - i),
                                             range(1, abs(int(k)) + 1), 1)
+        else:
+            if k.is_number:
+                if x.is_integer and x.is_negative:
+                    return S.Zero
```

Comment:
Do we need the `is_number` check? Should this not also evaluate for `Symbol('x', integer=True, negative=True)`?

Reply:
Oh I see that the check is for `k.is_number` rather than for `x`. Still though what is the check for? Should this not also evaluate for symbolic k?

Reply:
It's a problem with inclusive / exclusive conditions. For generic (complex) `Symbol('k')`, if we don't check that `k` is number, we'll end up in this branch and `RisingFactorial(-1, k)` will return `0`.